FUNCZ-349 Define a JSON/YAML variant of the recipe (#46)
BlueBrain/sonata-extension: source/recipe.rst

.. _recipe:

Recipe description
==================

.. highlight:: json

This document specifies all parameters required to generate the connectome of a circuit.

Components of the recipe are used by TouchDetector and Functionalizer as noted.  An
example or a TouchDetector configuration may look like:

.. code:: yaml

   recipe:
     bouton_distances:
       min_distance: 0.1
       max_distance: 10.0
       region_gap: 5.0
     structural_spine_lengths:
       - mtype: L7_A
         spine_length: 5.0
       - mtype: L9_DKE
         spine_length: 12.3

.. _selection:

Generic pathway selection
-------------------------

> Some of the recipe rules may apply only to certain "pathways".  These rules select the

Reviewer: ```suggestion
One can create rules that only apply to certain ``pathways``.
```
}

   =========== ============ =========== ============

Components
----------

``bouton_distances``
^^^^^^^^^^^^^^^^^^^^

*Optional*, used by Functionalizer.

Minimum distances for synapses, as measured along the branch length, starting at the soma.
Synapses with less than the specified distance will be removed by Functionalizer.

.. table::

   =============================== =========== ===
   Property                        Requirement Description
   =============================== =========== ===
   ``excitatory_synapse_distance`` Optional    The minimum distance from the soma for a synapse of post-synaptic excitatory cells. Defaults to 5µm.
   ``inhibitory_synapse_distance`` Optional    The minimum distance from the soma for a synapse of post-synaptic inhibitory cells. Defaults to 25µm.
   =============================== =========== ===

``bouton_interval``
^^^^^^^^^^^^^^^^^^^

*Required*, used by TouchDetector.

Distances used when transforming touch regions into synapse candidates.

.. table::

   =============================== =========== ===
   Property                        Requirement Description
   =============================== =========== ===

>    ``min_distance``                Mandatory   The minimum distance at which two synapse candidates are placed from each other.

Reviewer: and what the unit is?

Author: µm. one could comment generically that all distances are in µm unless otherwise mentioned :thinking:

Reviewer: Cool, I'd probably put that everywhere so it's hard to miss, and now one ever asks.

property name, and the standard deviation by appending ``SD`` to the
property name:

.. table::

   ==================================== =========== ===
   Property                             Requirement Description
   ==================================== =========== ===

>    ``gsyn``                             Mandatory   The peak conductance (in nS) for a single synaptic contact, following a Gamma distribution.

Reviewer: would it make sense for these to match the names in SONATA?

Author: it for sure would be less confusing and remove another layer of mapping

Reviewer: Do you think we're (kinda?) making a new format, that is our chance to change it?

Author: Yes, will work on that

Reviewer: @eleftherioszisis has noticed that the names are also used in NEXUS; maybe hold off a bit to see how much many changes that will require.

Reviewer 2: I would also prefer if we all used the SONATA spec as our one and only truth. :)

Author: Based on @eleftherioszisis' feedback I've started on the renaming - it will make things less confusing and more straightforward within the functionalizer, and introduces a 1:1 correspondence between the recipe and the output, so that users will not have to wonder which settings affect which output column.
